addons/stock/models/stock_quant.py

```diff
@@ -748,7 +752,7 @@ def _update_reserved_quantity(self, product_id, location_id, quantity, lot_id=No
 
         if float_compare(quantity, 0, precision_rounding=rounding) > 0:
             # if we want to reserve
-            available_quantity = self._get_available_quantity(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
+            available_quantity = sum(quants.filtered(lambda q: float_compare(q.quantity, 0, precision_rounding=rounding) > 0).mapped('quantity')) - sum(quants.mapped('reserved_quantity'))
```

Comment:
Why do you need this? You couldn't use allow_negative in the _get_available_quantity?

Reply:
Okay I get it. It's in the purpose of taking available quants if they exist and skip the counter balance of negative